galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/QueryParameters.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.framework.api.ras.internal;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.lang.reflect.Array;
+import java.time.*;
+import java.time.format.DateTimeParseException;
+
+import javax.servlet.http.HttpServletResponse;
+import static javax.servlet.http.HttpServletResponse.*;
+
+import static dev.galasa.framework.api.ras.internal.ServletErrorMessage.*;
+
+
+/**
+ * A wrapper for a map of query parameters, with methods for extracting
+ * and verifying the parameter contents.
+ */
+public class QueryParameters {
+    
+    Map<String,String[]> params ;
+
+	/**
+	 * @param paramMap The query parameters, probably coming from the request.
+	 */
+    public QueryParameters( Map<String,String[]> paramMap ) {
+        this.params = paramMap;
+    }
+	
+	/**
+	 * 
+	 * @param queryParameterName The query parameter
+	 * @param defaultValue
+	 * @return
+	 * @throws InternalServletException when there are multiple instances of the key, with error GAL5006
+	 */
+    public String getSingleString(String queryParameterName, String defaultValue ) throws InternalServletException {
+		String returnedValue = defaultValue ;
+		String[] paramValues = params.get(queryParameterName);
+		if (paramValues != null && paramValues.length >0) {
+
+			if (paramValues.length >1) {
+				ServletError error = new ServletError(
+					GAL5006_INVALID_QUERY_PARAM_DUPLICATES,
+					queryParameterName);
+				throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
+			}
+
+			String firstOccurrance = paramValues[0];
+			if( firstOccurrance != null ) {
+				String trimmedFirstOccurrance = firstOccurrance.trim();
+				if ( !(trimmedFirstOccurrance.isEmpty())) {
+					returnedValue = trimmedFirstOccurrance;
+				}
+			}
+		}
+		return returnedValue;
+	}
+
+	// /**
+	//  * 
+	//  * @param queryParameterName The query parameter
+	//  * @param defaultValues
+	//  * @return
+	//  */
+    public List<String> getMultipleString(String queryParameterName, List<String> defaultValues ) {
+		List<String> returnedValues = defaultValues ;
+		String[] paramValues = params.get(queryParameterName);
```

Comment:
What if there are duplicate query parameters, both being lists ? They should be combined ?

Reply:
putting two query parameters of the same name would not work, as it won't work from the params.get() function.